[DOC] Add documentation for alerting dashboards and visualizations (#4024)
opensearch-project/documentation-website: _observing-your-data/alerting/dashboards-alerting.md

---
layout: default
title: Integrating alerts with Dashboard 
parent: Alerting
nav_order: 50
---

> # Integrating alerts with Dashboard

Author: This is the Dashboard interface/application in OpenSearch Dashboards. Revised title.

> Introduced 2.8

Reviewer: Change to 2.9 please

{: .label .label-purple }

Create, manage, and take action on your alerts in a single, consolidated view and identify and resolve issues quickly. Use the **Dashboard** interface to:

- Set up, add, and adjust rules and conditions that trigger alerts and notifications.

> - Create graphs that show trends and patterns and build intuitive dashboards to stay on top of important metrics and data points in real time.

Reviewer 2: ```suggestion
- Create graphs that show trends and patterns and build intuitive dashboards to stay informed of important metrics and data points in real time.
```

- Monitor your alerts in one place with at-a-glance views.

> <img src="{{site.url}}{{site.baseurl}}/images/dashboards/alerting-viz.png" alt="Example alerting visualization" width="800" height="800">

Reviewer 2: Reference the image in the text that precedes it.

> <img src="{{site.url}}{{site.baseurl}}/images/dashboards/alerting-viz.png" alt="Example alerting visualization" width="800" height="800">

Author: @lezzago Do we also need to update this screenshot (as we've done for AD)?

Reviewer: Provided a new image in the `alerting-viz.png` comment.

Before getting started, you must have:

- Installed OpenSearch and OpenSearch Dashboards version 2.8 or later. See [Installing OpenSearch]({{site.url}}{{site.baseurl}}/install-and-configure/install-opensearch/index/).

> - Installed Alerting and Notifications Dashboards plugins. See [Managing OpenSearch Dashboards plugins]({{site.url}}{{site.baseurl}}/install-and-configure/install-dashboards/plugins/) to get started.

Reviewer 2: ```suggestion
- Installed the Alerting and Notifications Dashboards plugins. See [Managing OpenSearch Dashboards plugins]({{site.url}}{{site.baseurl}}/install-and-configure/install-dashboards/plugins/) to get started.
```

## Configuring admin settings

> You can only access, create, or manage alerts for resources for which you have permissions. Access to alerting dashboards and visualizations is controlled by OpenSearch and OpenSearch Dashboards privileges, and you can manage the settings in **Stack Management**. Access is enabled by default and appears as a feature in the **Stack Management** > **Advanced Settings** > **Visualizations** window. If the setting is disabled, it does not appear in this window. The setting is disabled at the cluster level through the `opensearch-dashboards.yml` file.

Reviewer 3: The setting "can be" disabled at the cluster level through the `opensearch-dashboards.yml` file

## General requirements for alerting visualizations

> Alerting visualizations are displayed as time-series charts that give you a snapshot of the alert, alert status, last updated time, and reason for the alert. You can show up to 10 metrics on your chart, and each series can be shown as a line in the chart.

Reviewer 2: ```suggestion
Alerting visualizations are displayed as time-series charts that give you a snapshot of the alert, alert status, last updated time, and reason for the alert. You can display up to 10 metrics on your chart, and each series can be shown as a line on the chart.
```

Keep in mind the following requirements when setting up or creating alerting visualizations:

> - Must be a [Vizlib line chart](https://community.vizlib.com/support/solutions/articles/35000107262-vizlib-line-chart-introduction)

Reviewer 2: It would be preferable for these to be structured as complete sentences.

Author: Revised

> - Must contain at least Y-axis metric aggregation

Reviewer 2: ```suggestion
- Must contain at least a Y-axis metric aggregation
```

> - Must be a [Vizlib line chart](https://community.vizlib.com/support/solutions/articles/35000107262-vizlib-line-chart-introduction)

Reviewer 3: I think we should be specific about what OpenSearch chart types are supported as Vizlib is likely not something that someone would know - 
<img width="402" alt="image" src="https://github.com/opensearch-project/documentation-website/assets/98365234/bc8c5e7a-a3bb-4bbb-bcb9-b1366305fa9e">

Reviewer 3: @ohltyler

Reviewer 3: We also need to call out what aggregations are supported (average(), count(), sum(), min(), or max()).

Author: @ohltyler Please confirm the viz types and aggregations supported in this feature.

> - Must contain one Y-axis metric aggregation

Reviewer: Can change to: `Must contain at least Y-axis metric aggregation`.
We allow multiple different metric aggregations.

> - Must not have non-Y-axis metric aggregation types

Author: @lezzago Is line 39 accurate?

Reviewer: Yes

> - Must use date histogram aggregation type for the X-axis bucket

Reviewer 2: ```suggestion
- Must use the date histogram aggregation type for the X-axis bucket
```

> - Must have X-axis on the bottom

Reviewer 2: "an" or "the" X-axis? The bottom of the...?

- Must define one X-axis aggregation bucket
- Must have a valid time-based X-axis

## Creating alerting monitors with Dashboard

> By default, when you begin to create the alert monitor workflow using the Dashboard app, you are presented with a menu-driven interface. This interface provides a range of options, displayed in full screen, pop-up, pull-down, or dropdown, to define the metrics to monitor, set thresholds, customize triggers that automate workflows, and generate actions when conditions are met. Currently, you can create query-level monitors only.

Reviewer 3: Query-level is referred to as 'per query' in documentation. Let's adjust for consistency. https://opensearch.org/docs/latest/observing-your-data/alerting/monitors/#monitor-types

> By default, when you begin to create the alert monitor workflow using the Dashboard interface, you are presented with a menu-driven interface. This interface provides a range of options that are displayed in full screen, in pop-ups, in pull-downs, or dropdowns. They allow you to define the metrics that can be monitored, set thresholds, customize triggers that automate workflows, and generate actions when conditions are met. Currently, you can create [per query monitors]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/monitors/#monitor-types) only.

Reviewer 2: ```suggestion
By default, when you begin to create the alert monitor workflow using the Dashboard interface, you are presented with a menu-driven interface. This interface provides a range of options that are displayed in full screen, in pop-ups, in pull-downs, or in dropdowns. They allow you to define the metrics that can be monitored, set thresholds, customize triggers that automate workflows, and generate actions when conditions are met. Currently, you can only create [per query monitors]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/monitors/#monitor-types).
```

#### To create an alerting monitor 

1. Choose **Dashboard** from the OpenSearch Dashboards main menu.
2. From the **Dashboards** window, select **Create** and then choose **Dashboard**.
3. Select **Add an existing**, then select the appropriate alerting visualization from the **Add panels** list. The visualization is added to the dashboard.
4. From the visualization panel, choose the ellipsis icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/ellipsis-icon.png" class="inline-icon" alt="ellipsis icon"/>{:/}). 
5. From the **Options** menu, select **Add alerting monitor**.

> 6. Input information for **Monitor details** and **Triggers**  

Reviewer 4: ```suggestion
6. Input information for **Monitor details** and **Triggers**.
```

7. Choose **Create monitor**. The monitor is added to the visualization.  

> An example of the create monitor steps is shown in the following screenshot.

Reviewer 2: ```suggestion
An example of these steps is shown in the following screenshot.
```

<img src="{{site.url}}{{site.baseurl}}/images/dashboards/create-monitor-menu.png" alt="Create monitor interface" width="400" height="400">

## Associating monitors

> You can associate existing monitors with a visualization using the Dashboard app instead of the plugin page, giving you a single interface to add, view, and edit monitor data.

Reviewer 3: Users can associated monitors which are not specific to per query monitors. This is another benefit of being able to associate a monitor.

Author: @brijos To confirm my understanding, can both existing monitors and monitor types other than per query monitors be associated with a visualization?

> You can associate certain monitor types with a visualization using the Dashboard interface instead of the plugin page, giving you a single interface to add, view, and edit monitor data.

Reviewer 2: ```suggestion
You can associate certain monitor types with a visualization using the Dashboard interface instead of the plugin page, giving you a single interface through which to add, view, and edit monitor data.
```

> #### To associate a monitor

Reviewer 2: This heading reads a bit oddly because it's used as a procedure heading above but as a section heading here (no sentence follows it in the preceding instance).

> You can associate certain monitors with a visualization using the Dashboard interface instead of the plugin page, giving you a single interface to add, view, and edit monitor data. You can associate existing monitors and monitors not specific to a per query monitor.

Reviewer 4: "You can associate existing monitors and monitors not specific to a per query monitor."
Is there another way to say, "... not specific to a query monitor"? I'm not sure what that means.
"... as well as monitors of other types, such as a per bucket or per document monitor" ?

Author: Revised

> Continuing with the alerting visualization and dashboard created in the preceding section, associate an existing monitor with a visualization by following these steps. 

Reviewer 2: ```suggestion
Continuing with the alerting visualization and dashboard created in the preceding section, associate an existing monitor with a visualization by following these steps: 
```

> Continuing with the alerting visualization and dashboard created in the preceding tutorial, associate an existing monitor with a visualization by following these steps. 

Reviewer 4: "... preceding section" ? (rather than tutorial)

1. From the visualization panel, choose the ellipsis icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/ellipsis-icon.png" class="inline-icon" alt="ellipsis icon"/>{:/}).
2. Select **Associated monitors**.
3. From the **Select monitor to associate** dropdown menu, select the monitor. Only eligible monitors are listed in the dropdown menu. 
4. View the monitor's basic information. To view comprehensive details, select **View monitor page** to open the Alerting plugin page.

> 5. Select **Associate monitor**. An existing monitor is associated to the visualization.

Reviewer 2: ```suggestion
5. Select **Associate monitor**. An existing monitor is associated with the visualization.
```

## Exploring alerting monitor details

> Once you've created or associated alerting monitors, verify the monitor is generating the alerts and explore alert details by following these steps:

Reviewer 2: ```suggestion
Once you've created or associated alerting monitors, verify that the monitor is generating alerts and explore alert details by following these steps:
```

1. Open the alerting dashboard. Alerts are indicated on the visualization with a triangle icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/dashboards/triangle-icon.png" class="inline-icon" alt="triangle icon"/>{:/}). 

> 2. Hover over a triangle to view high-level data, such as number of alerts. To investigate alert details, select the triangle icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/dashboards/triangle-icon.png" class="inline-icon" alt="triangle icon"/>{:/}) to open a flyout with more detailed monitor information. Alternatively, select the ellipsis icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/ellipsis-icon.png" class="inline-icon" alt="ellipsis icon"/>{:/}) in the visualization panel and choose **View events**.

Reviewer 4: "Hover over a triangle to view high-level data, such as number of alerts."
Specify that we're hovering over a "triangle icon" to avoid the possibility a reader thinks there might be another kind of three-sided geometric shape to look for?

3. Select the ellipsis icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/ellipsis-icon.png" class="inline-icon" alt="ellipsis icon"/>{:/}), then **Alerting** > **Associated monitors**.

> 4. Choose an alerting monitor from the list. Information such as history, alerts, and associated visualizations are shown within the visualization panel.

Reviewer 2: ```suggestion
4. Choose an alerting monitor from the list. Information such as history, alerts, and associated visualizations is shown within the visualization panel.
```

> 5. Explore unlinking or editing a monitor. 

Reviewer 2: We've used "explore" a lot already. Can we use something like "Learn how to unlink or edit a monitor"?

Author: Revised

>    1. Unlink a monitor from the visualization by selecting the link icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/dashboards/link-icon.png" class="inline-icon" alt="link icon"/>{:/}) under **Actions**. This unlinks the monitor from the visualization only; it does not delete the monitor.

Reviewer 2: ```suggestion
   1. Unlink a monitor from the visualization by selecting the link icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/dashboards/link-icon.png" class="inline-icon" alt="link icon"/>{:/}) under **Actions**. This only unlinks the monitor from the visualization; it does not delete the monitor.
```

   2. Edit the monitor's metrics by selecting the edit icon ({::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/dashboards/edit-icon.png" class="inline-icon" alt="edit icon"/>{:/}).

## Next steps

> - [Learn more about Dashboard](https://opensearch.org/docs/latest/dashboards/dashboard/index/)

Reviewer 2: ```suggestion
- [Learn more about the Dashboard application](https://opensearch.org/docs/latest/dashboards/dashboard/index/).
```

> - [Learn more about alerting](https://opensearch.org/docs/latest/observing-your-data/alerting/index/)

Reviewer 2: ```suggestion
- [Learn more about alerting](https://opensearch.org/docs/latest/observing-your-data/alerting/index/).
```
